core/src/test/scala/integration/kafka/api/TransactionsExpirationTest.scala

```diff
@@ -105,6 +105,36 @@ class TransactionsExpirationTest extends KafkaServerTestHarness {
     }
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testTransactionAfterProducerIDExpires(quorum: String): Unit = {
+    producer.initTransactions()
+
+    // Start and then abort a transaction to allow the producer ID to expire
+    producer.beginTransaction()
+    producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic1, 0, "2", "2", willBeCommitted = false))
+    producer.send(TestUtils.producerRecordWithExpectedTransactionStatus(topic2, 0, "4", "4", willBeCommitted = false))
+    producer.abortTransaction()
+
+    // Wait for the producer ID to expire
+    Thread.sleep(1000)
```

Comment:
I'm not sure I follow -- we still want the producer ID to expire right?

Reply:
Yes, we do. Could we use waitUntilTrue and look into the producer state manager instead of using sleep? That would be a stronger assertion if that is possible.

Reply:
got it. thanks for clarifying

Reply:
np.

Reply:
I looked into this more and realized that this test actually checked before the producer IDs were added. 
To fix this, I added a wait until to make sure the producer IDs are added and it works.

However, now I realize that there is a hard-coded `ProducerIdExpirationCheckIntervalMs = 10 * 60 * 1000` in LogManager which makes this a bit harder to test. (The equivalent check interval for transactions is a configuration -- `KafkaConfig.TransactionsRemoveExpiredTransactionalIdCleanupIntervalMsProp`.) I don't think we necessarily need a new configuration here, but I need to think about how I can test this in an integration test. Maybe we don't need to do this here.

Reply:
Yeah, this is exactly why I don't like those sleeps in test. In the end, you don't know if you are lucky or if the conditions you wanted really were met.